bot/cogs/assignable_roles_cog.py

```diff
@@ -23,21 +23,19 @@ def __init__(self, bot) -> None:
     @ext.long_help('Lists all roles that have been marked as assignable in this server')
     @ext.short_help('Defines custom assignable roles')
     @ext.example('roles')
-    async def roles(self, ctx, *, input_role: str = None) -> None:
-        if input_role is None:
+    async def roles(self, ctx, input_roles: commands.Greedy[discord.Role] = None, bad_args: str = None) -> None:
+        if input_roles is None or bad_args:
```

Comment:
the check for bad args needs to come at the end, you can have a valid role in between invalid ones

Reply:
make the default for input_roles an empty list then you can check bad args at the end